### PR TITLE
Clean lwaftr object files

### LINKED ISSUE #678: `make clean` should also delete .o files in tests dir?
Example:

```
$ make clean
RM        snabb obj bin testlog programs.inc
$ ls program/lwaftr/tests/data/*.o
program/lwaftr/tests/data/icmp_on_fail.o
```

Since the `.o` file remains, you can get an error like this if you update snabb, rebuild, and run a test:

```
program/lwaftr/tests/data/icmp_on_fail.conf: loading compiled configuration from program/lwaftr/tests/data/icmp_on_fail.o
core/main.lua:26: incompatible version: 8192
stack traceback:
        core/main.lua:149: in function <core/main.lua:147>
        [C]: in function 'error'
        core/main.lua:26: in function 'assert'
        lib/yang/binary.lua:406: in function 'load_compiled_data'
        lib/yang/yang.lua:66: in function 'load_compiled'
        lib/yang/yang.lua:94: in function 'load_lwaftr_config'
        program/lwaftr/run/run.lua:134: in function 'run'
        program/lwaftr/lwaftr.lua:17: in function 'run'
        core/main.lua:67: in function <core/main.lua:43>
        [C]: in function 'xpcall'
        core/main.lua:211: in main chunk
        [C]: at 0x00453700  
        [C]: in function 'pcall'
        core/startup.lua:3: in main chunk
        [C]: in function 'require'
        [string "require "core.startup""]:1: in main chunk
```

### PR DESCRIPTION
**Pull request by dpino**
Fixes #678 cc @takikawa.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -209,7 +209,14 @@ obj/doc/snabb.epub: obj/doc/snabb.markdown
 
 CLEAN = snabb obj bin testlog programs.inc
 
-clean:
+clean_programs:
+	@(for d in program/*/; do \
+		if [[ -f "$$d/Makefile" ]]; then \
```
Comment (kbara) [inline src/Makefile]:
``` 
% make clean_programs
/bin/sh: 2: [[: not found
```

**Comment by dpino** [comment]
Fixed comment. I reworked the script a bit. Example of execution:

```
$ make clean
CLEAN     program/lwaftr/
RM        ./tests/data/no_icmp.o
RM        ./tests/data/icmp_on_fail.o
RM        ./tests/data/vlan.o
RM        doc/snabb-lwaftr.*
CLEAN     program/snabbvmx/
RM        doc/snabbvmx.*
RM        snabb obj bin testlog programs.inc
```

**Comment by kbara** [comment]
That's looking a lot better, but snabbvmx's Makefile needs better cleaning:

```
% find program -name '*.o'
program/snabbvmx/tests/end-to-end/data/binding_table.txt.s.o
program/snabbvmx/tests/end-to-end/data/snabbvmx-lwaftr-xe1.o
program/snabbvmx/tests/end-to-end/data/vlan/binding_table.txt.s.o
program/snabbvmx/tests/end-to-end/data/vlan/snabbvmx-lwaftr-xe1.o
```

**Comment by dpino** [comment]
OK, I added the a clean_object_files rule in `snabbvmx/Makefile`. Ideally the patch should be ported to `lwaftr-snabbvmx` branch but while `lwaftr-snabbvmx` doesn't catch up with `lwaftr` branch, I would only do that for critical bugs. This is more like an enhancement.

**Review by teknico — APPROVED** [review]
LGTM

**Comment by kbara** [comment]
LGTM too. @dpino , will you merge this?

**Pull request merged**